SdkClientException: Unable to execute HTTP request: null

https://sentry.io/zerocracy/farm/issues/464802246/

```
ClosedChannelException: None
    at sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:110)
    at sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:147)
    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:65)
    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:109)
    at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:103)
...
(45 additional frame(s) were not displayed)

SdkClientException: Unable to execute HTTP request: null
    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1113)
    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1063)
    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:743)
    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
...
(14 additional frame(s) were not displayed)

Unable to execute HTTP request: null
```

@amihaiemil one more: 

```
java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/cost/make_payment.groovy
	at com.zerocracy.farm.reactive.StkGroovy.process(StkGroovy.java:125)
	at com.zerocracy.farm.StkSafe.process(StkSafe.java:81)
	at com.zerocracy.farm.reactive.StkSmart.process(StkSmart.java:66)
	at com.zerocracy.farm.reactive.Brigade.process(Brigade.java:82)
	at com.zerocracy.farm.reactive.Brigade.lambda$apply$0(Brigade.java:64)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:58)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
	at org.cactoos.iterator.Filtered.hasNext(Filtered.java:90)
	at org.cactoos.iterator.LengthOf.lambda$new$0(LengthOf.java:52)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:58)
	at org.cactoos.scalar.NumberEnvelope.doubleValue(NumberEnvelope.java:113)
	at org.cactoos.iterable.LengthOf.lambda$new$0(LengthOf.java:51)
	at org.cactoos.scalar.NumberEnvelope.lambda$new$1(NumberEnvelope.java:73)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.scalar.UncheckedScalar.value(UncheckedScalar.java:58)
	at org.cactoos.scalar.NumberEnvelope.intValue(NumberEnvelope.java:98)
	at com.zerocracy.farm.reactive.Brigade.apply(Brigade.java:67)
	at com.zerocracy.farm.reactive.Brigade.apply(Brigade.java:37)
	at org.cactoos.func.IoCheckedBiFunc.lambda$apply$0(IoCheckedBiFunc.java:60)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:61)
	at com.zerocracy.farm.reactive.DefaultFlush.process(DefaultFlush.java:98)
	at com.zerocracy.farm.reactive.DefaultFlush.lambda$exec$0(DefaultFlush.java:67)
	at org.cactoos.func.FuncOf.lambda$new$3(FuncOf.java:88)
	at org.cactoos.func.FuncOf.apply(FuncOf.java:104)
	at org.cactoos.func.IoCheckedFunc.lambda$apply$0(IoCheckedFunc.java:59)
	at org.cactoos.scalar.IoCheckedScalar.value(IoCheckedScalar.java:70)
	at org.cactoos.func.IoCheckedFunc.apply(IoCheckedFunc.java:59)
	at org.cactoos.func.IoCheckedProc.exec(IoCheckedProc.java:57)
	at com.zerocracy.pm.Claims.take(Claims.java:143)
	at com.zerocracy.farm.reactive.DefaultFlush.exec(DefaultFlush.java:66)
	at com.zerocracy.farm.reactive.DefaultFlush.exec(DefaultFlush.java:44)
	at com.zerocracy.farm.reactive.AsyncFlush.lambda$exec$2(AsyncFlush.java:118)
	at com.jcabi.log.VerboseCallable.call(VerboseCallable.java:173)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at com.jcabi.log.VerboseThreads$Wrap.run(VerboseThreads.java:221)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.zerocracy.farm.reactive.StkGroovy.process(StkGroovy.java:115)
	... 41 more
Caused by: com.amazonaws.SdkClientException: Unable to execute HTTP request: null
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1113)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1063)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:743)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:717)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4247)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4194)
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1738)
	at com.amazonaws.services.s3.transfer.internal.UploadCallable.uploadInOneChunk(UploadCallable.java:133)
	at com.amazonaws.services.s3.transfer.internal.UploadCallable.call(UploadCallable.java:125)
	at com.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:143)
	at com.amazonaws.services.s3.transfer.internal.UploadMonitor.call(UploadMonitor.java:48)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: java.nio.channels.ClosedChannelException
	at sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:110)
	at sun.nio.ch.FileChannelImpl.read(FileChannelImpl.java:147)
	at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:65)
	at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:109)
	at sun.nio.ch.ChannelInputStream.read(ChannelInputStream.java:103)
	at org.apache.commons.io.input.ProxyInputStream.read(ProxyInputStream.java:100)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:82)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:82)
	at com.amazonaws.util.LengthCheckInputStream.read(LengthCheckInputStream.java:107)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:82)
	at com.amazonaws.services.s3.internal.MD5DigestCalculatingInputStream.read(MD5DigestCalculatingInputStream.java:128)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at com.amazonaws.internal.SdkBufferedInputStream.read(SdkBufferedInputStream.java:76)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:82)
	at com.amazonaws.event.ProgressInputStream.read(ProgressInputStream.java:180)
	at com.amazonaws.internal.SdkFilterInputStream.read(SdkFilterInputStream.java:82)
	at org.apache.http.entity.InputStreamEntity.writeTo(InputStreamEntity.java:140)
	at com.amazonaws.http.RepeatableInputStreamRequestEntity.writeTo(RepeatableInputStreamRequestEntity.java:160)
	at org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:156)
	at org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:162)
	at org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:238)
	at com.amazonaws.http.protocol.SdkHttpRequestExecutor.doSendRequest(SdkHttpRequestExecutor.java:63)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:123)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:271)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:55)
	at com.amazonaws.http.apache.client.impl.SdkHttpClient.execute(SdkHttpClient.java:72)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1235)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1055)
	... 17 more
```

--- a/src/main/java/com/zerocracy/farm/S3Item.java
+++ b/src/main/java/com/zerocracy/farm/S3Item.java
@@ -20,6 +20,8 @@ import com.amazonaws.services.s3.model.ObjectMetadata;
 import com.jcabi.log.Logger;
 import com.jcabi.s3.Ocket;
 import com.zerocracy.Item;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -28,6 +30,11 @@ import java.nio.file.attribute.FileTime;
 import java.util.Date;
 import java.util.concurrent.atomic.AtomicBoolean;
 import lombok.EqualsAndHashCode;
+import org.cactoos.io.InputOf;
+import org.cactoos.io.LengthOf;
+import org.cactoos.io.OutputTo;
+import org.cactoos.io.TeeInput;
+import org.cactoos.scalar.IoCheckedScalar;
 
 /**
  * Item in S3.
@@ -35,8 +42,9 @@ import lombok.EqualsAndHashCode;
  * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
  * @since 0.1
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
-@EqualsAndHashCode(of = { "ocket", "temp" })
+@EqualsAndHashCode(of = {"ocket", "temp"})
 final class S3Item implements Item {
 
     /**
@@ -125,7 +133,24 @@ final class S3Item implements Item {
             final ObjectMetadata meta = new ObjectMetadata();
             final long start = System.currentTimeMillis();
             meta.setContentLength(this.temp.toFile().length());
-            this.ocket.write(Files.newInputStream(this.temp), meta);
+            final ByteArrayOutputStream buff = new ByteArrayOutputStream();
+            Logger.debug(
+                this,
+                "buffered file %s %f bytes",
+                this.temp,
+                new IoCheckedScalar<>(
+                    new LengthOf(
+                        new TeeInput(
+                            new InputOf(this.temp),
+                            new OutputTo(buff)
+                        )
+                    )
+                ).value()
+            );
+            this.ocket.write(
+                new ByteArrayInputStream(buff.toByteArray()),
+                meta
+            );
             Files.setLastModifiedTime(
                 this.temp,
                 FileTime.fromMillis(
@@ -171,5 +196,4 @@ final class S3Item implements Item {
         return !remote.equals(local)
             || this.temp.toFile().length() != meta.getContentLength();
     }
-
 }

--- a/src/main/java/com/zerocracy/farm/S3Item.java
+++ b/src/main/java/com/zerocracy/farm/S3Item.java
@@ -21,7 +21,6 @@ import com.jcabi.log.Logger;
 import com.jcabi.s3.Ocket;
 import com.zerocracy.Item;
 import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -30,11 +29,8 @@ import java.nio.file.attribute.FileTime;
 import java.util.Date;
 import java.util.concurrent.atomic.AtomicBoolean;
 import lombok.EqualsAndHashCode;
+import org.cactoos.io.BytesOf;
 import org.cactoos.io.InputOf;
-import org.cactoos.io.LengthOf;
-import org.cactoos.io.OutputTo;
-import org.cactoos.io.TeeInput;
-import org.cactoos.scalar.IoCheckedScalar;
 
 /**
  * Item in S3.
@@ -133,22 +129,12 @@ final class S3Item implements Item {
             final ObjectMetadata meta = new ObjectMetadata();
             final long start = System.currentTimeMillis();
             meta.setContentLength(this.temp.toFile().length());
-            final ByteArrayOutputStream buff = new ByteArrayOutputStream();
-            Logger.debug(
-                this,
-                "buffered file %s %f bytes",
-                this.temp,
-                new IoCheckedScalar<>(
-                    new LengthOf(
-                        new TeeInput(
-                            new InputOf(this.temp),
-                            new OutputTo(buff)
-                        )
-                    )
-                ).value()
-            );
             this.ocket.write(
-                new ByteArrayInputStream(buff.toByteArray()),
+                new ByteArrayInputStream(
+                    new BytesOf(
+                        new InputOf(this.temp)
+                    ).asBytes()
+                ),
                 meta
             );
             Files.setLastModifiedTime(

